### PR TITLE
JAX-WS: Add ManagedExecutor fat tests

### PR DESCRIPTION
**Pull request by neuwerk**
This PR adds a new fat bucket `io.openliberty.jaxws.executor_fat` that tests our integration with the `concurrent` features, using the `ManagedScheduledExecutorService` which can be set on a JAX-WS dynamic proxy client.

### CODE REVIEW

**Comment by neuwerk** [comment]
#build

File: dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java [hunk 1]
```diff
@@ -0,0 +1,811 @@
...
+        while(!response.isDone()) {
+            // Wait for response to finish
+        }
+        
+        // Get the returned string from the Web Service endpoint
+        actualResponse = (String) response.get();
+       
+        // Assert that the response is what we expect
+        if (actualResponse != null) {
+            assertTrue("The testSimpleNonManagedClient test failed: the actual response - " + actualResponse + " did not equal the expected response - " + expectedResponse,
```
Comment (BerksanAtes) [inline dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java]: Test name should be testAsyncClientWithResponse

File: dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java [hunk 2]
```diff
@@ -0,0 +1,811 @@
...
+        
+        // Instantiate a AsyncHandler object 
+        AsyncHandler handler = new AsyncHandler() {
+            public void handleResponse(Response response) {
+              try {
+                // Get the response from the web service 
+                actualResponse = (String) response.get(1000, TimeUnit.MILLISECONDS);
+                
+                // do the assert in the handler code
+                assertTrue("The testSimpleNonManagedClient test failed: the actual response - " + actualResponse + " did not equal the expected response - " + expectedResponse,
```
Comment (BerksanAtes) [inline dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java]: Test name should be testAsyncClientWithAsyncHandler

File: dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java [hunk 3]
```diff
@@ -0,0 +1,811 @@
...
+        
+        // Use the Excecutor set on the client to submit the simpleHelloCall CallBack
+        Future<String> future = ((ManagedScheduledExecutorService) scheduledServiceClient.getExecutor()).submit(simpleHelloCall);
+        
+        // Get the String response from the Web Service Impl
+        actualResponse = future.get(TIMEOUT_NS, TimeUnit.NANOSECONDS);
+        
+        // Assert that the response is what we expect
+        if (actualResponse != null) {
+            assertTrue("The testSimpleNonManagedClient test failed: the actual response - " + actualResponse + " did not equal the expected response - " + expectedResponse,
```
Comment (BerksanAtes) [inline dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java]: Test name should be testClientSubmitCallViaManagedExecutor

File: dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java [hunk 4]
```diff
@@ -0,0 +1,811 @@
...
```
Comment (BerksanAtes) [inline dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java]: Test name should be testClientSubmitCallViaManagedExecutorThreadAndInstance

File: dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java [hunk 5]
```diff
@@ -0,0 +1,811 @@
...
+        while(!response.isDone()) {
+            // Wait for response to finish
+        }
+       
+        // Get the string respone from the Web Service async Response object
+        actualResponse = (String) response.get();
+
+        
+        if (actualResponse != null) {
+            assertTrue("The testSimpleNonManagedClient test failed: the actual response - " + actualResponse + " did not equal the expected response - " + expectedResponse,
```
Comment (BerksanAtes) [inline dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java]: Test name should be testAsyncClientSubmitCallViaManagedExecutorThreadAndInstance

File: dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java [hunk 6]
```diff
@@ -0,0 +1,811 @@
...
+        
+        // Instantiate a AsyncHandler object 
+        AsyncHandler handler = new AsyncHandler() {
+            public void handleResponse(Response response) {
+              try {
+                // Get the response from the web service 
+                actualResponse = (String) response.get(1000, TimeUnit.MILLISECONDS);
+                
+                // do the assert in the handler code
+                assertTrue("The testSimpleNonManagedClient test failed: the actual response - " + actualResponse + " did not equal the expected response - " + expectedResponse,
```
Comment (BerksanAtes) [inline dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java]: Test name should be testAsyncClientWithAsyncHandlerViaManagedExecutorWrappedCallThreadAndInstance

File: dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java [hunk 7]
```diff
@@ -0,0 +1,811 @@
...
+               // Get the response from the web service 
+                actualResponse = (String) response.get(1000, TimeUnit.MILLISECONDS);
+
+                LOG.info(method + " Response from inside simpleHelloCall: " + actualResponse);
+                
+                // Add response to the Queue
+                result.add(actualResponse);
+                
+                // Do the assert
+                assertTrue("The testSimpleNonManagedClient test failed: the actual response - " + actualResponse + " did not equal the expected response - " + expectedResponse,
```
Comment (BerksanAtes) [inline dev/io.openliberty.jaxws.executor_fat/test-applications/simpleservice/src/io/openliberty/jaxws/fat/JAXWSExecutorTestServlet.java]: Test name should be testAsyncClientWithAsyncHandlerWithoutManagedExecutor

**Review by BerksanAtes — DISMISSED** [review]
Looks good to me!

**Comment by neuwerk** [comment]
#build

**Comment by neuwerk** [comment]
#build

**Comment by neuwerk** [comment]
#build

**Review by BerksanAtes — APPROVED** [review]
Looks good to me!

**Comment by neuwerk** [comment]
#libby

**Pull request merged**